src/react-vite/03-intro-to-jsx/intro-to-jsx.md

```diff
@@ -55,62 +68,216 @@ Take the below HTML and convert it to JSX:
 
 @diff ../../../exercises/react-vite/02-setting-up-your-environment/solution/src/App.css ../../../exercises/react-vite/03-intro-to-jsx/01-problem/src/App.css only
 
 @diff ../../../exercises/react-vite/02-setting-up-your-environment/solution/src/index.css ../../../exercises/react-vite/03-intro-to-jsx/01-problem/src/index.css only
 
+Next, you'll need to pull the image from the assets package by adding the following import into `src/App.tsx`.
+
+```tsx
+import HeroImage from 'place-my-order-assets/images/homepage-hero.jpg'
+```
+
+If you inspect the type on the import, you'll notice it is a string. It's actually just the URL for the image. You can use the image import by setting it as the `src` of an image element like `src={HeroImage}`
+
 ### Verify
 
-TODO
+You should have already installed Vitest in the previous section. If you haven't done so already, create a test file at `src/App.test.tsx` and copy the following tests into your test file.
 
 @diff ../../../exercises/react-vite/02-setting-up-your-environment/solution/src/App.test.tsx ../../../exercises/react-vite/03-intro-to-jsx/01-problem/src/App.test.tsx only
 
+Run your tests and verify the results
+
+```shell
+npm run test
+```
+
+Note that we won't be able to write an automated test that verifies the desired styles and images appear as desired, so you will have to test those manually.
+
 ### Solution
 
 <details>
 <summary>Click to see the solution</summary>
 
-TODO
+✏️ Update **src/App.tsx**
 
 @diff ../../../exercises/react-vite/03-intro-to-jsx/01-problem/src/App.tsx ../../../exercises/react-vite/03-intro-to-jsx/01-solution/src/App.tsx only
 
 </details>
 
-## Objective 2
-
-TODO
+## Objective 2 - Loops in JSX
 
 ### Key concepts
 
-TODO
+- JSX Interpolation
+- Loops in JSX
 
-#### Concept 1
+## JSX Interpolation
 
-TODO
+JSX is dynamic. You can easily insert values from variables and objects into your JSX as we did with the image URL in the previous section.
 
-#### Concept 2
+```jsx
+const name = 'Bitovi';
 
-TODO
+<div className="button primary">Welcome to {name}!</div>;
+```
 
-### Requirements
+In the code above, use the `{name}` syntax to tell JSX that to render the value stored in the `name` variable (i.e. `"Bitovi"`) into our view.
 
-TODO
+You can take this a step further by interpolating multiple values, and using JavaScript functions to transform data on the fly. Anything that goes inside `{}` is executed as normal JavaScript. These are the same rules as the brackets on a prop: any JavaScript expression is valid inside the curly brackets.
 
-### Setup
+```jsx
+const person = {
+  name: 'mike',
+  profession: 'programmer',
+};
+
+<div className="button primary">
+  <h1>Hi I'm {person.name.toUpperCase()}!</h1>
+  <p>I'm a {person.profession} living in Philadelphia</p>
+</div>;
+```
+
+@highlight 7,8
+
+### JSX is JavaScript
+
+Remember, JSX is simply an alternative syntax for normal JavaScript&mdash;it is not magic. This means that you can use JSX as a normal value.
```

Comment:
I love an em dash 😍 FYI it’s _shift + option + hyphen (-)_ if you want to type “—” and _option + hyphen (-)_ if you want an en dash (–).